### DISCUSSION TITLE
Issue with CIDRAM IP Tracking and Data Disappearance

### DISCUSSION THREAD

**Discussion by borborich** [CIDRAM/CIDRAM#522] [Q&A] [answered]
I am currently facing an issue with CIDRAM, and I'm seeking assistance from the community to help me resolve it.

The problem I'm encountering is related to the tracking of IP addresses and the unexpected disappearance of data in CIDRAM. I have integrated CIDRAM into my OpenCart website, and while it initially works as expected, I've noticed that the time stamps for CIDRAM's execution are periodically updated, and previously tracked IP addresses seem to vanish from the CIDRAM database.

Here are some additional details:

I have included CIDRAM's execution within my OpenCart's index.php file as follows:
```
<?php
// Version
define('VERSION', '2.3.0.2.3');

require_once('vault/loader.php');
(new \CIDRAM\CIDRAM\Core())->protect();
// Other OpenCart code follows
```
I suspect that there might be configurations or settings within CIDRAM that control the retention of IP addresses and data, but I haven't been able to pinpoint the exact cause of this behavior.
I would greatly appreciate any insights, guidance, or suggestions from the CIDRAM community on why this might be happening and how I can ensure that CIDRAM retains IP tracking data consistently.

If you have any experience or expertise related to CIDRAM, or if you've encountered a similar issue before, please share your insights. Your assistance will be highly valuable in resolving this matter.

Thank you in advance for your help, and I look forward to any feedback or advice you can provide.

**Comment by Maikuolan** [comment, marked as answer]
Because IP tracking stores its data in the cache, it's possible that problems with caching could cause IP tracking data to disappear (assuming we mean all of it disappearing entirely, all at once). When the IP tracking data disappears, does CIDRAM's statistics and cache data also disappear? If so, problems with caching is the most likely culprit. If not, then possibly, something unrelated might be the cause.

When there are problems with caching, experimenting with the configuration directives under **`supplementary_cache_options`** (i.e., to change to a different cache mechanism) can sometimes solve the problem. Unfortunately, because the reliability of the various available cache mechanisms can vary significantly between different hosting environments (depending on what's available at those hosting environments, how everything is configured at the server, etc), it's difficult to say "this will always be the best" or "this will always be the worst" for any of them (which is one of the reasons why multiple such mechanisms are supported like that in the first place, allowing most users to hopefully be able to take advantage of whatever works best for them).

In terms of specific, *individual* IP tracking entries disappearing (i.e., expiring), **`signatures➡default_tracktime`** controls the default duration for IP tracking entries.
